src/compiler/checker.ts

```diff
@@ -1363,13 +1363,15 @@ const enum IntrinsicTypeKind {
     Lowercase,
     Capitalize,
     Uncapitalize,
+    NoInfer,
 }
 
 const intrinsicTypeKinds: ReadonlyMap<string, IntrinsicTypeKind> = new Map(Object.entries({
     Uppercase: IntrinsicTypeKind.Uppercase,
     Lowercase: IntrinsicTypeKind.Lowercase,
     Capitalize: IntrinsicTypeKind.Capitalize,
     Uncapitalize: IntrinsicTypeKind.Uncapitalize,
+    NoInfer: IntrinsicTypeKind.NoInfer,

```

Comment:
Slightly related question: should `ThisType` become an intrinsic type as well (it is kinda intrinsic by nature but it's not defined as such in the typedefs)? Or is it better to not touch it at all?

Reply:
Obviously this shouldn't be touched as part of this PR - I just take this opportunity to ask a question about it ;p

Reply:
We want to leave `ThisType` as is. An `intrinsic` declaration doesn't really say anything about the internal representation of the type, and we already have a perfectly good solution for with the current interface-based declaration.